src/Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/TimeDependentTripleGaussian.hpp

```diff
@@ -0,0 +1,203 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/DampingFunction.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace domain::FunctionsOfTime {
+class FunctionOfTime;
+}  // namespace domain::FunctionsOfTime
+/// \endcond
+
+namespace GeneralizedHarmonic::ConstraintDamping {
+/*!
+ * \brief A sum of three Gaussians plus a constant, where the Gaussian widths
+ * are scaled by a domain::FunctionsOfTime::FunctionOfTime.
+ *
+ * \details The function \f$f\f$ is given by
+ * \f{align}{
+ * f = C + \sum_{\alpha=1}^3
+ * A_\alpha \exp\left(-\frac{(x-(x_0)_\alpha)^2}{w_\alpha^2(t)}\right).
+ * \f}
+ * Input file options are: `Constant` \f$C\f$, `Amplitude[1-3]`
+ * \f$A_\alpha\f$, `Width[1-3]` \f$w_\alpha\f$, `Center[1-3]
+ * `\f$(x_0)_\alpha\f$, and `FunctionOfTimeForScaling`, a string naming a
+ * domain::FunctionsOfTime::FunctionOfTime in the domain::Tags::FunctionsOfTime
+ * that will be passed to the call operator. The function takes input
+ * coordinates \f$x\f$ of type `tnsr::I<T, VolumeDim, Fr>`, where `T` is e.g.
+ * `double` or `DataVector`, `Fr` is a frame (e.g. `Frame::Inertial`), and
+ * `VolumeDim` is the dimension of the spatial volume. The Gaussian widths
+ * \f$w_\alpha\f$ are scaled by the inverse of the value of a scalar
+ * domain::FunctionsOfTime::FunctionOfTime \f$f(t)\f$ named
+ * `FunctionOfTimeForScaling`: \f$w_\alpha(t) = w_\alpha / f(t)\f$.
+ */
+template <size_t VolumeDim, typename Fr>
+class TimeDependentTripleGaussian : public DampingFunction<VolumeDim, Fr> {
+ public:
+  template <size_t GaussianNumber>
+  struct Gaussian {
+    static constexpr Options::String help = {
+        "Parameters for one of the Gaussians."};
+    static std::string name() noexcept {
+      return "Gaussian" + std::to_string(GaussianNumber);
+    };
+  };
+  struct Constant {
+    using type = double;
+    static constexpr Options::String help = {"The constant."};
+  };
+
+  template <typename Group>
+  struct Amplitude {
+    using group = Group;
+    using type = double;
+    static constexpr Options::String help = {"The amplitudes of Gaussian."};
```

Comment:
`"The amplitude of " + group::name()` and similarly for the other options

Reply:
Unless there's something @nilsdeppe and I missed, this is not possible. 

The solution you propose does not compile, because the + operator is not defined for const char* strings like Option::Strings. Trying to make it work by first creating a std::string, adding it to a std::string initialized by group, and then converting back to a const char* fails because that result requires runtime evaluation (can't be a constexpr)